projects/gstreamer/build.sh

```diff
@@ -0,0 +1,115 @@
+echo "CXXFLAGS" $CXXFLAGS
+export CFLAGS="$CFLAGS $BASEFLAGS"
+export CXXFLAGS="$CXXFLAGS $BASEFLAGS"
+export LDFLAGS="$SANITIZER_FLAGS $COVERAGE_FLAGS"
+PREFIX=$WORK/prefix
+PLUGIN_DIR=$PREFIX/lib/gstreamer-1.0
+export PKG_CONFIG_PATH=$PREFIX/lib/pkgconfig
+mkdir -p $PREFIX
+cd $WORK
+
+for i in orc gstreamer gst-plugins-base;
+do
+    mkdir -p $i
+    cd $i
+    $SRC/$i/autogen.sh --prefix=$PREFIX --disable-shared --enable-static --disable-examples --disable-gtk-doc --disable-introspection --enable-static-plugins --disable-gst-tracer-hooks
+    make -j$(nproc)
+    make install
+    cd ..
+done
+
+#finally build the binary \o/
+BUILD_CFLAGS="$CFLAGS `pkg-config --static --cflags glib-2.0 gstreamer-1.0 gstreamer-pbutils-1.0 gstreamer-video-1.0 gstreamer-audio-1.0 gstreamer-app-1.0 orc-0.4`"
+
+# List of dependencies libraries we grab from pkg-config
+# Should also include dependencies of dependencies (ex: libvorbis depends on libogg)
+
+PKG_DEPS="glib-2.0 gstreamer-1.0 gstreamer-pbutils-1.0 gstreamer-video-1.0 gstreamer-audio-1.0 orc-0.4 \
+  gstreamer-riff-1.0 gstreamer-tag-1.0 gstreamer-app-1.0 zlib \
+  ogg vorbis vorbisenc theoraenc theoradec theora"
+
+# List of all plugins to include
+PLUGINS="$PLUGIN_DIR/libgstcoreelements.a \
+       $PLUGIN_DIR/libgsttypefindfunctions.a \
+       $PLUGIN_DIR/libgstplayback.a \
+       $PLUGIN_DIR/libgstapp.a \
+       $PLUGIN_DIR/libgstvorbis.a \
+       $PLUGIN_DIR/libgsttheora.a \
+       $PLUGIN_DIR/libgstogg.a"
+
+# We want to statically link everything, except for shared libraries that are present on
+# the base image. Those need to be specified beforehad and explicitely linked dynamically
+# If any of the static dependencies require a pre-installed shared library, you need
+# to add that library to the following list
+PREDEPS_LDFLAGS="-Wl,-Bdynamic -ldl -lm -pthread -lrt -lpthread"
+
+# The libraries we want to statically link to
+# This includes dependencies of the gst plugins
+#BUILD_LDFLAGS="$LDFLAGS `pkg-config --static --libs $PKG_DEPS` -Wl,-static -lpcre "
+BUILD_LDFLAGS="$LDFLAGS -Wl,-static `pkg-config --static --libs $PKG_DEPS`"
+
+echo
+echo "PREDEPS_LDFLAGS" $PREDEPS_LDFLAGS
+echo
+echo "BUILD_LDFLAGS" $BUILD_LDFLAGS
+echo
+
+echo $CXX $CXXFLAGS $BUILD_CFLAGS \
+      -v -Wl,--verbose \
+      $SRC//gst-discoverer.c \
+      $PLUGINS \
+      $LIB_FUZZING_ENGINE \
+      -o gst-discoverer \
+      $PREDEPS_LDFLAGS \
+      $BUILD_LDFLAGS \
+      -Wl,-Bdynamic
+
+echo
+echo ">>>> BUILDING gst-discoverer.o"
+echo
+
+$CC $CFLAGS $BUILD_CFLAGS -c $SRC/gst-discoverer.c -o $SRC/gst-discoverer.o
+
+echo
+echo ">>>> LINKING"
+echo
+
+$CXX $CXXFLAGS \
+      -v -Wl,--verbose \
+      -o $OUT/gst-discoverer \
+      $PREDEPS_LDFLAGS \
+      $SRC/gst-discoverer.o \
+      $PLUGINS \
+      $BUILD_LDFLAGS \
+      $LIB_FUZZING_ENGINE \
+      -Wl,-Bdynamic
+
+echo
+echo ">>>> Downloading OGG corpus"
+echo
+
+cd $OUT
+wget -nd https://people.freedesktop.org/~bilboed/gst-discoverer_seed_corpus.zip
```

Comment:
this is better be done in Dockerfile

Reply:
Downloading the corpus ? Or putting it in $OUT/ ?

Reply:
Downloading, e.g. https://github.com/google/oss-fuzz/blob/master/projects/libreoffice/Dockerfile

Reply:
ok, fixing it